src/Symfony/Component/HttpKernel/EventListener/ServiceResetListener.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * Clean up services between requests.
+ *
+ * @author Alexander M. Turek <me@derrabus.de>
+ */
+class ServiceResetListener implements EventSubscriberInterface
+{
+    private $services;
+    private $resetMethods;
+
+    public function __construct(\Traversable $services, array $resetMethods)
```

Comment:
I would remove the `\Traversable` typehint to support any iterable (phpdoc at first; `iterable` typehint when merging up)

Reply:
The listener is meant to be used with an iterator of initialized services. Right now, I don't see a use case for allowing arrays here.